app-text/mupdf/mupdf-1.25.2.ebuild

```diff
@@ -67,6 +67,12 @@ src_prepare() {
 
 	append-cflags "-DFZ_ENABLE_JS=$(usex javascript 1 0)"
 
+	if ! use jpeg2k; then
+		append-cflags "-DFZ_ENABLE_JPX=0"
+		sed -i '/lopenjp2/d' Makerules || die
+		sed -i '/openjpeg.h/d' 'source/fitz/encode-jpx.c' || die
```

Comment:
> semi-supported by upstream: https://github.com/ArtifexSoftware/mupdf/commit/6975007f314a4c392e2cb616e7c3297ed167c2a7
> 
> requires a bit of sed patching to remove unconditional openjpeg.h include and linking.

I would think that if we want to properly support this, you should submit a patch to upstream that makes this include respect the macro as well. I can understand that handling the linking is harder to do if it cannot be done by a macro (and upstream seems adamantly opposed to real build systems) but reducing the downstream cognitive burden feels correct here...

Reply:
> I would think that if we want to properly support this, you should submit a patch to upstream that makes this include respect the macro as well.

Okay, done: https://github.com/ArtifexSoftware/mupdf/pull/60

> I can understand that handling the linking is harder to do if it cannot be done by a macro

I don't see an obvious way to do this. If you have any ideas feel free to shoot. Otherwise I think the sed patching should be fine.

Reply:
Yeah no I got no ideas either. :)

As long as we do the best to reduce our sed debt where there's an obvious way to do it, we should be good for now I think. Thank you for the upstream PR.